core/src/main/scala/org/apache/spark/sql/streaming/StreamSqlHelper.scala

```diff
@@ -68,7 +68,7 @@ object StreamSqlHelper {
     val encoder = ExpressionEncoder[A]()
     val schema = encoder.schema
     val logicalPlan = LogicalDStreamPlan(schema.toAttributes,
-      stream.map(encoder.toRow))(ssc)
+      stream.map(encoder.toRow(_).copy()))(ssc)
     new SchemaDStream(ssc, logicalPlan)
   }
 
```

Comment:
Doesn't the other one below also require it? Perhaps not exercised in any existing test.